master/internal/command/command_service.go

```diff
@@ -0,0 +1,232 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/rm"
+	"github.com/determined-ai/determined/master/internal/task"
+	"github.com/determined-ai/determined/master/pkg/logger"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+// DefaultCmdService is the global command service singleton.
+var DefaultCmdService *CommandService
+
+// CommandService tracks the different NTSC commands in the system.
+type CommandService struct {
```

Comment:
I'd prefer all the methods for this struct in the same file. I was trying to vet the locking and it is hard without this.

Also, i think there are some locking optimizations we can do, but let's in a follow up.